conan/tools/cmake/toolchain.py

```diff
@@ -473,8 +473,8 @@ def context(self):
         # To find the generated cmake_find_package finders
         # TODO: Change this for parameterized output location of CMakeDeps
         find_package_prefer_config = "ON"  # assume ON by default if not specified in conf
-        prefer_config = self._conanfile.conf["tools.cmake.cmaketoolchain:find_package_prefer_config"]
-        if prefer_config is not None and prefer_config.lower() in ("false", "0", "off"):
+        prefer_config = self._conanfile.conf.get("tools.cmake.cmaketoolchain:find_package_prefer_config")
+        if prefer_config in (False, 0) or (isinstance(prefer_config, str) and prefer_config.lower() in ("false", "off")):
```

Comment:
A bit dirty. I'd probably add a ``get(.., check_type=bool)`` smart converter or a ``conf.get_bool()``, or something that implements our "boolean" conf convention for ``off, false, 0`` etc